lib/rules/no-descending-specificity/index.js

```diff
@@ -137,17 +142,27 @@ function lastCompoundSelectorWithoutPseudoClasses(selectorNode) {
 	const nodesAfterLastCombinator = nodesByCombinator[nodesByCombinator.length - 1];
 
 	assert(nodesAfterLastCombinator);
-	const nodesWithoutPseudoClasses = nodesAfterLastCombinator
-		.filter((node) => {
-			return (
-				node.type !== 'pseudo' ||
-				node.value.startsWith('::') ||
-				pseudoElements.has(node.value.replace(/:/g, ''))
-			);
-		})
-		.join('');
 
-	return nodesWithoutPseudoClasses.toString();
+	const nodesWithoutPseudoClasses = nodesAfterLastCombinator.filter((node) => {
+		return (
+			isCssModulesGlobal(node) ||
+			node.type !== 'pseudo' ||
+			node.value.startsWith('::') ||
+			pseudoElements.has(node.value.replace(/:/g, ''))
+		);
+	});
+
+	if (nodesWithoutPseudoClasses.length === 0) return undefined;
+
+	return nodesWithoutPseudoClasses.join('');
+}
+
+/**
+ * @param {import('postcss-selector-parser').Node} node
+ * @returns {boolean}
+ */
+function isCssModulesGlobal(node) {
```

Comment:
[previously @jeddy3 wanted to not handle `:global()`](https://github.com/stylelint/stylelint/issues/5157#issuecomment-782668548)

related: #443